pkg/client/oci/oci.go

```diff
@@ -0,0 +1,91 @@
+package oci
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/jetstack/version-checker/pkg/api"
+	"github.com/jetstack/version-checker/pkg/version/semver"
+)
+
+type Client struct{}
+
+func New() (*Client, error) {
+	return &Client{}, nil
+}
```

Comment:
I think we could construct a `*remote.Puller` here and use it for the remote operations in `Tags`. This will cache auth across different calls, reducing the potential number of requests made for each operation.

See:
https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/puller.go